### ISSUE TITLE
bug(datepicker): selected date range is not visible using Firefox with high contrast mode

### ISSUE THREAD

**Issue by zarend** [angular/components#23034]
#### Reproduction

Use StackBlitz to reproduce your issue: 
- Components: https://stackblitz.com/edit/components-issue-2xkpr2

Steps to reproduce:
1. Click on date picker toggle
2. (Datepicker dialog opens)
3. Select a start date and and end date
4. (Start date and end date of highlighted)
 
 
#### Expected Behavior

Start date, end date, and all dates in-between are highlighted.

#### Actual Behavior

Cells of selected dates between the selected start date and end date look same as not selected dates. I can't tell them apart.

<img width="1187" alt="Screen Shot 2021-06-21 at 1 46 24 PM" src="https://user-images.githubusercontent.com/7720245/122829591-0bd26100-d29c-11eb-951f-da863bde7773.png">

#### Environment

- Angular: 12
- CDK/Material: 12
- Browser(s): Firefox 89.0.1 (64-bit)
- Operating System (e.g. Windows, macOS, Ubuntu): macOS Big Sur version 11.4 (20F71)
